grafana/grafana
# logs: merge log-lines-based and full-range histogram

we need to do certain changes to the histogram-visualization (add an enable/disable button and so on), but it's complicated because we have the old-histogram and the new-histogram too.

this pull-request merges the visualization of both histograms into one `<ExploreGraph/>` react component, this should allow us to do changes only at a single place.

how to test:
1. `make devenv sources=loki`
2. test that new-histogram works:
    1. go to explore, run `{place="moon"}`
    2. you should get a gray histogram. this is the new-histogram. you can now zoom "into" the time-range in the histogram using the mouse, and you will get a `reload log volume` button
3. test that old-histogram works:
    1. find the `getLogsVolumeDataProvider` method at https://github.com/grafana/grafana/blob/gabor/logs-merge-histograms/public/app/plugins/datasource/loki/datasource.ts#L106 , and modify it: add `return undefined;` as the first line. this will make grafana use the old-histogram for Loki
    2. go to explore, run `{place="moon"}`
    3. you should get a colored histogram, with a message explaining that this is the old-histogram
